torchbench.py

```diff
@@ -730,18 +776,43 @@ def main():
 
     experiment = functools.partial(experiment, args, model_iter_fn)
 
+    if args.speedup_fx2trt_fp16:
+        cos_similarity = True
+    else:
+        cos_similarity = False
+
     if output_filename:
         output_filename = os.path.join(torchdynamo.config.base_dir, output_filename)
 
     if args.minimum_call_count:
         torchdynamo.config.minimum_call_count = args.minimum_call_count
-
     if args.only:
         for device in args.devices:
             try:
                 device, name, model, example_inputs = load_model(
                     device, args.only, args.training, args.check_accuracy
                 )
+                # torchbench changed the default precison=fp16 on torchvision net
+                if name in (
+                    "alexnet",
+                    "resnet18",
+                    "resnet50",
+                    "mobilenet_v2",
+                    "mnasnet1_0",
+                    "squeezenet1_1",
+                    "shufflenetv2_x1_0",
+                    "vgg16",
+                    "resnext50_32x4d",
+                ):
+                    try:
+                        assert (
+                            args.speedup_fx2trt_fp16 == True
+                        ), "Do not test vision models in fp32 mode"
+                    except NotImplementedError:
+                        continue  # not supported benchmark implementation
```

Comment:
Unreachable?

Reply:
Not really, just want to stop fp32 run here. Otherwise, line 923 will throw error. The reason is that TRT output is fp32 and reference model output is fp16. They can not be compared.

Reply:
The only thing inside the `try:` is an `assert`   How will we get a NotImplementedError?

Reply:
I made it complicated. Will fix it.